src/components/modals/buy-modal/index.js

```diff
@@ -35,161 +34,106 @@ bitcoin.initEccLib(ecc);
 const BuyModal = ({ show, handleModal, utxo, onSale, nostr }) => {
   const {
     nostrOrdinalsAddress,
-    nostrPaymentsAddress,
+    nostrPaymentAddress,
     ordinalsPublicKey,
     paymentPublicKey,
   } = useWallet();
-  const [isBtcInputAddressValid, setIsBtcInputAddressValid] = useState(true);
-  const [destinationBtcAddress, setDestinationBtcAddress] =
-    useState(nostrPaymentsAddress);
+  const [
+    isOrdinalDestinationAddressValid,
+    setIsOrdinalDestinationAddressValid,
+  ] = useState(true);
+  const [ordinalsDestinationAddress, setOrdinalsDestinationAddress] =
+    useState(nostrOrdinalsAddress);
   const [isOnBuy, setIsOnBuy] = useState(false);
-  const [selectedUtxos, setSelectedUtxos] = useState([]);
-  const [deezyPsbtPopulate, setDeezyPsbtPopulate] = useState(null);
   const [buyTxId, setBuyTxId] = useState(null);
 
   const [isMounted, setIsMounted] = useState(true);
   const showDiv = useDelayUnmount(isMounted, 500);
   const { bitcoinPrice } = useBitcoinPrice({ nostrOrdinalsAddress });
 
-  const updatePayerAddress = async (address) => {
-    try {
-      let deezyPsbt;
-      if (!deezyPsbtPopulate) {
-        const { data } = await axios.post(
-          `https://api${
-            TESTNET ? "-testnet" : ""
-          }.deezy.io/v1/ordinals/psbt/populate`,
-          {
-            psbt: nostr.content, // (hex or base64)
-            ordinal_receive_address: address,
-          }
-        );
-
-        const { psbt } = data;
-        deezyPsbt = bitcoin.Psbt.fromHex(psbt, {
-          network: NETWORK,
-        });
-
-        setDeezyPsbtPopulate(data);
-      } else {
-        const { psbt } = deezyPsbtPopulate;
-        deezyPsbt = bitcoin.Psbt.fromHex(psbt, {
-          network: NETWORK,
-        });
+  const populateDeezyPsbt = async () => {
+    const {
+      data: { psbt, id },
+    } = await axios.post(
+      `https://api${
+        TESTNET ? "-testnet" : ""
+      }.deezy.io/v1/ordinals/psbt/populate`,
+      {
+        psbt: nostr.content, // (hex or base64)
+        ordinal_receive_address: ordinalsDestinationAddress,
       }
+    );
 
-      let requiredFee = null;
```

Comment:
why are you removing this?

Reply:
please put it back, this handles an issue in which before we do the utxo selection the fee might bee too high,so we need to re-calc the required utxos required to pay for the fee